src/porepy/compositional/compositional_mixins.py

```diff
@@ -1095,25 +1092,39 @@ def create_mixture(self) -> None:
 
         self.set_components_in_phases(components, phases)
 
-        self.fluid_mixture = FluidMixture(components, phases)
+        self.fluid = Fluid(components, phases)
 
     def get_components(self) -> list[Component]:
-        """Method to return a list of modelled components."""
-        raise CompositionalModellingError(
-            "Call to mixin method. Define components by overriding this method."
-        )
+        """Method to return a list of modelled components.
+
+        The default implementation takes the user-provided or default fluid constants
+        found in the model ``params`` and returns a single component.
+
+        Override this method via mixin to provide a more complex component context for
+        the :attr:`fluid`.
+
+        """
+        # Should be available after SolutionStrategy.set_materials()
+        # Getting the user-passed fluid constants to create the default fluid component
```

Comment:
This is confusing. Why are the user-passed constants private?

Reply:
The user-passed constants are now used to create the 1 component for the default case of a 1-phase, 1-component fluid. I wanted to hint to developers to not mess with this objects, which I stored in `params` until the `model.fluid` is created.

Related to your previous comment https://github.com/pmgbergen/porepy/pull/1252#discussion_r1822352597

I can add more documentation to explain this, but it belongs to the inner workings and is (hopefully) only exposed to developers digging pretty deep. I am also open for a better mechanism to transfer the fluid constants from the `params` dicts provided in the instantiation of the solution strategy, to this point in the chain `prepare_simulation() -> set_materials() -> create_fluid()`

Reply:
Or maybe I should just use `params['material_constants']['fluid']` ...

Reply:
I removed the ominous `_default_fluid_constans` and rely solely on `self.params['material_constants']['fluid']`, also in the FluidMixin class.